src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -1442,6 +1447,7 @@ public static async ValueTask<FormulaValue> Summarize(EvalVisitor runner, EvalVi
                 var showColumnsArgs = new List<FormulaValue>() { row.Value };
                 showColumnsArgs.AddRange(stringArgs);
 
+                // We call ShowColumns to keep only the columns that are part of the group by.
                 var keyRecord = await ShowColumns(runner, context, IRContext.NotInSource(FormulaType.Build(irContext.ResultType._type.ToRecord())), showColumnsArgs.ToArray()).ConfigureAwait(false);
                 var key = keyRecord.ToExpression();

```

Comment:
#2778 - other issue, we should probably not be calling .ToExpression.

Reply:
This will be addressed at #2778